core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -764,6 +769,19 @@ private[spark] class TaskSetManager(
     maybeFinishTaskSet()
   }
 
+  private[scheduler] def markPartitionCompleted(partitionId: Int): Unit = {
+    partitionToIndex.get(partitionId).foreach { index =>
+      if (!successful(index)) {
+        tasksSuccessful += 1
+        successful(index) = true
+        if (tasksSuccessful == numTasks) {
+          isZombie = true
+        }
+        maybeFinishTaskSet()
```

Comment:
is this line needed? We will call `maybeFinishTaskSet()` at the end of `handleSuccessfulTask`

Reply:
I think you're right, its not needed, its called when the tasks succeed, fail, or are aborted, and when this called while that taskset still has running tasks, then its a no-op, as it would fail the `runningTasks == 0` check inside `maybeFinishTaskSet()`.

do you think its worth removing?  I'm fine either way.

Reply:
it's too minor. If we touch this file again, let's remove it. Otherwise maybe not bother about it.